sw/device/exts/common/flash_crt.S

```diff
@@ -117,9 +117,9 @@ init_array_loop_end:
   li   a1, 0x0  // argv = NULL
   call main
 
-  // Load `kDeviceStopAddress` into `t0`. If it is non-zero, write the return
-  // value from `main` (in `a0`) into it, otherwise, skip to `wfi` loop.
-  lw t0, kDeviceStopAddress
+  // Load `kDeviceTestStatusAddress` into `t0`. If it is non-zero, write the
+  // return value from `main` (in `a0`) into it, otherwise, skip to `wfi` loop.
+  lw t0, kDeviceTestStatusAddress
   beqz t0, wfi_loop
   sw a0, 0(t0)
```

Comment:
The most likely return value of `main` is 1 or 0, which aren't valid test status values. We can perhaps update this code to avoid writing any test status, and just entering a `wfi` loop on exit of main, unless writing that value won't cause an issue.

Reply:
Honestly it probably just makes sense to make main noreturn in the fullness of time...